src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -53,18 +53,14 @@ var validUTCDateTime = function (dateTimeComposition){
 };
 
 // check the string is valid
-var validString = function (dateTimeString) {
+var validDatetimeString = function (dateTimeString) {
     if (!dateTimeString || dateTimeString.toString() === '')
         return false;
-    return true;
-};
-
-// check the characters of string are digits
-var validDigitString = function (dateTimeString) {
+    // datetime format in the spec: YYYY[MM[DD[HH[MM[SS[.S[S[S[S]]]]]]]]][+/-ZZZZ],
     var ds = dateTimeString.toString();
-    if (!/^(\d+)$/.test(ds))
-        throw `Contains non-numeric characters in ${ds}`;
-    return ds;
+    if (!/^(\d{4}(\d{2}(\d{2}(\d{2}(\d{2}(\d{2}(\.\d+)?)?)?)?)?)?((-|\+)\d{1,4})?)$/.test(ds))
```

Comment:
Looking at the spec the time zone has to be 4 digits. I think the time zone check should be d{4} instead of d{1,4}.

Reply:
Ranvijay has noticed that another spec for the ccda is written as follows. Here I want to support 1~4 digits for timezone, because it may exist, and I don't want to throw exceptions too strictly.
```
<xs:simpleType name="ts">
<xs:annotation>
<xs:documentation> A quantity specifying a point on the axis of natural time. A point in time is most often represented as a calendar expression. </xs:documentation>
</xs:annotation>
<xs:restriction base="xs:string">
<xs:pattern value="[0-9]{1,8}|([0-9]{9,14}|[0-9]{14,14}\.[0-9]+)([+\-][0-9]{1,4})?"/>
</xs:restriction>
</xs:simpleType>
```